packages/protocol-contracts/contracts/ERC20Custody.sol

```diff
@@ -0,0 +1,214 @@
+     */
+    function renounceTSSAddressUpdater() external {
+        if (msg.sender != TSSAddressUpdater) {
+            revert InvalidTSSUpdater();
+        }
+        if (TSSAddress == address(0)) {
+            revert ZeroAddress();
+        }
+        TSSAddressUpdater = TSSAddress;
+    }
+
+    /**
+     * @dev Pause custody operations.
+     */
+    function pause() external {
+        if (paused) {
+            revert IsPaused();
+        }
+        if (msg.sender != TSSAddressUpdater) {
+            revert InvalidTSSUpdater();
+        }
+        if (TSSAddress == address(0)) {
+            revert ZeroAddress();
+        }
+        paused = true;
+        emit Paused(msg.sender);
+    }
+
+    /**
+     * @dev Unpause custody operations.
+     */
+    function unpause() external {
+        if (!paused) {
+            revert NotPaused();
+        }
+        if (msg.sender != TSSAddressUpdater) {
+            revert InvalidTSSUpdater();
+        }
+        paused = false;
+        emit Unpaused(msg.sender);
+    }
+
+    /**
+     * @dev Whitelist asset
+     * @param asset, ERC20 asset
+     */
+    function whitelist(IERC20 asset) external {
+        if (msg.sender != TSSAddress) {
+            revert InvalidSender();
+        }
+        whitelisted[asset] = true;
+        emit Whitelisted(asset);
+    }
+
+    /**
+     * @dev Unwhitelist asset
+     * @param asset, ERC20 asset
+     */
+    function unwhitelist(IERC20 asset) external {
+        if (msg.sender != TSSAddress) {
+            revert InvalidSender();
+        }
+        whitelisted[asset] = false;
+        emit Unwhitelisted(asset);
+    }
+
+    /**
+     * @dev Deposit asset amount to recipient with message that encodes additional zetachain evm call or message.
+     * @param recipient, recipient address
+     * @param asset, ERC20 asset
+     * @param amount, asset amount
+     * @param message, bytes message or encoded zetechain call.
+     */
+    function deposit(bytes calldata recipient, IERC20 asset, uint256 amount, bytes calldata message) external {
+        if (paused) {
+            revert IsPaused();
+        }
+        if (!whitelisted[asset]) {
+            revert NotWhitelisted();
+        }
+        if (address(zeta) != address(0)) {
+            zeta.transferFrom(msg.sender, TSSAddress, zetaFee);
+        }
+        asset.transferFrom(msg.sender, address(this), amount);
+        emit Deposited(recipient, asset, amount, message);
+    }
+
+    /**
+     * @dev Deposit asset amount to recipient with message that encodes additional zetachain evm call or message.
+     * @param recipient, recipient address
+     * @param asset, ERC20 asset
+     * @param amount, asset amount
+     * @param message, bytes message or encoded zetechain call.
+     */
+    function depositFeeOnTransfer(bytes calldata recipient, IERC20 asset, uint256 amount, bytes calldata message) external {
```

Comment:
@aldapp7 what's the difference between this one and the deposit() above?

Reply:
Hmm, sorry, it was supposed to be modified method for fee on token transfer, but now I doubt that we really want to support token with fees on transfers, complicates it and also don't see major protocols doing that despite audit claims.